### DISCUSSION TITLE
How do i set a category on a new company?

### DISCUSSION THREAD

**Discussion by PPetterssonSO** [SuperOffice/typescript-for-superoffice#17] [General]
Hi,
We have started to really look into Crm typescript, so there will be quite alot of questions comming your way.

So, lets start simple, using createDefaultContactEntityAsync, how do i set the category?
It reffers to SO.Category, but, where can i find that?

What im trying to achive in crmscript: 
NSContactEntity y;
NSCategory x;
x.SetId(123);
y.SetCategory(x);

**Comment by ejfasting** [comment]
Hello @Nawksen,

Exciting that you are starting to test typescript for SuperOffice!

This is based on the same functionality as the [@superoffice/webapi](https://www.npmjs.com/package/@superoffice/webapi?activeTab=code) package. So in this case I will refer you to our [sample on docs](https://docs.superoffice.com/en/api/netserver/web-services/howto/company/create-contact-webapi-agents.html), as it (hopefully) explains what you are after. 

```typescript
const agent = soApi.getContactAgent();
let entity = await agent.createDefaultContactEntityAsync();
entity.category.id = 1;
```

Have a SUPER day!

//Eivind

**Comment by PPetterssonSO** [comment]
So, no more NSCategory, im so, so, so very happy!
Thanks!

Regards
Pär
